backend/tournesol/views/previews/recommendations.py

```diff
@@ -276,14 +276,31 @@ def draw_no_recommendations_text(self, image: Image.Image):
             fill=COLOR_BROWN_FONT,
         )
 
+    def get_limit_offset(self, request):
+        try:
+            offset = max(0, int(request.GET["offset"]))
+        except (KeyError, ValueError):
+            offset = 0
+
+        try:
+            limit = min(3, int(request.GET["limit"]))
+        except (KeyError, ValueError):
+            limit = 3
+
+        if limit <= 0:
+            limit = 3
+
+        return offset, limit
+
     @method_decorator(cache_page_no_i18n(CACHE_RECOMMENDATIONS_PREVIEW))
     @extend_schema(exclude=True)
     def get(self, request, *args, **kwargs):
         upscale_ratio = self.upscale_ratio
 
         preview_image = Image.new("RGBA", (440 * upscale_ratio, 240 * upscale_ratio), "#FAFAFA")
 
-        recommendations = super().get_queryset()[:3]
+        offset, limit = self.get_limit_offset(request)
```

Comment:
(minor) To make it less error-prone
```suggestion
        offset, limit = self.get_offset_limit(request)
```

Reply:
Good catch!

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1736/commits/77553f2ad70f6f40b9913c3d15ea023588190709